File: Behavioral/Memento/Java/Editor.java
// Editor.java

public class Editor {
  private String content = "";

  public void type(String words) {
    content = words;
  }

  public EditorMemento save() {
    return new EditorMemento(content);
  }

  public void restore(EditorMemento memento) {
    if (memento != null) {
      content = memento.getContent();
    }
  }

  public String getContent() {
    return content;
  }
}

File: Behavioral/Memento/Cpp/Editor.cpp
// Editor.cpp

#include "Editor.h"

void Editor::type(const std::string& words) {
  content = words;
}

EditorMemento Editor::save() const {
  return EditorMemento(content);
}

void Editor::restore(const EditorMemento& memento) {
  content = memento.getContent();
}

std::string Editor::getContent() const {
  return content;
}
